File: Python/383_RansomNote.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        alphabet = [0] * 26
        for character in magazine:
            alphabet[ord(character) - ord('a')] += 1
        for character in ransomNote:
            index = ord(character) - ord('a')
            if alphabet[index] <= 0:
                return False
            alphabet[index] -= 1
        return True

File: Java/383_RansomNote.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] alphabet = new int[26];  //number of characters in alphabet
        int i;
        for(char c : ransomNote.toCharArray()){
            i = magazine.indexOf(c , alphabet[c-'a']);
            if(i == -1){
                return false;
            }
            alphabet[c - 'a'] = i + 1;
        }
        return true;
    }
}